core-api/src/main/java/com/optimizely/ab/Optimizely.java

```diff
@@ -153,6 +159,7 @@ public void close() {
         tryClose(eventProcessor);
         tryClose(eventHandler);
         tryClose(projectConfigManager);
+        NotificationRegistry.clearNotificationCenterRegistry();
```

Comment:
Is this safe to clear all notificationCenters? We can clear one for the current sdkKey. @msohailhussain also shared concern about memory leak when we close a notificationCenter, which may be addressed together when we clean it up.